patchwork-capabilities/src/main/java/net/minecraftforge/event/AttachCapabilitiesEvent.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Minecraft Forge, Patchwork Project
+ * Copyright (c) 2016-2019, 2019
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation version 2.1
+ * of the License.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+
+package net.minecraftforge.event;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+import com.google.common.collect.Lists;
+import com.google.common.collect.Maps;
+import net.minecraftforge.common.capabilities.ICapabilityProvider;
+import net.minecraftforge.eventbus.api.GenericEvent;
+
+import net.minecraft.util.Identifier;
+
+/**
+ * Fired whenever an object with Capabilities support {currently TileEntity/Item/Entity)
+ * is created. Allowing for the attachment of arbitrary capability providers.
+ *
+ * Please note that as this is fired for ALL object creations efficient code is recommended.
```

Comment:
That's frightening.

Reply:
Class was also copied from Forge. And I also don’t understand why Forge isn’t that slow with all these events